### PR TITLE
Record `accepted_at` and `declined_at`

### PR DESCRIPTION
**Pull request by duncanjbrown**
## Context

While working on the date widget at the top of applications in the Provider Interface (#945), we noticed that 

- `declined_at` was nil, even though `DeclineOffer` had been called on an application
- we weren't storing `accepted_at` whereas all other dates are

## Changes proposed in this pull request

- Add `accepted_at` to `ApplicationChoices`
- Set it in `AcceptOffer`
- Set `declined_at` in `DeclineOffer`
- Use `DeclineOffer` instead of manually calling `ApplicationStateChange`

## Link to Trello card

https://trello.com/c/Php2IR0H/1409-solve-incongruities-between-provider-and-candidate-interfaces
## Things to check

- [ ] This code doesn't rely on migrations in the same Pull Request — ⚠️  **IT DOES** ⚠️  but there are no offers in the system rn and no applications in an offerable state, so there is a very low risk of an offer being accepted, which is the code path which depends on the migration
- [X] API release notes have been updated if necessary
- [X] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/services/accept_offer.rb [hunk 1]
```diff
@@ -6,9 +6,10 @@ def initialize(application_choice:)
   def save!
     ActiveRecord::Base.transaction do
       ApplicationStateChange.new(@application_choice).accept!
+      @application_choice.update(accepted_at: Time.zone.now)
 
       other_application_choices_with_offers.each do |other_application_choice|
-        ApplicationStateChange.new(other_application_choice).decline!
+        DeclineOffer.new(application_choice: other_application_choice).save!
```
Comment (mnacos) [inline app/services/accept_offer.rb]: Much better! This, however, may trigger multiple Slack notifications for the automatic declines. Are we ok with that?

Comment (duncanjbrown) [reply app/services/accept_offer.rb]: I decided we were, but open to being convinced the other way. We could always flag it if it became annoying

Comment (duncanjbrown) [reply app/services/accept_offer.rb]: Oh wait we have the block which silences Slack... let me find it

**Comment by duncanjbrown** [comment]
@mnacos updated to not send slack notifications on the declines

**Pull request merged**